### ISSUE TITLE
Cray-internal Jenkins: change all e-mail senders in Jenkins and scripts to use authenticated TLS SMTP

### ISSUE THREAD

**Issue by awallace-cray** [chapel-lang/chapel#11233]
We have been warned that traditional Cray-internal SMTP services will be disabled ~soon~ Dec 1 2018.
This is more of an issue for Chapel's Cray-internal infrastructure than for Chapel itself. 
Several scripts in chapel.git/util/\*\* will need work.

**Comment by awallace-cray** [comment]
2018-10-12 
https://github.com/chapel-lang/chapel/pull/11360 -  send_email.py can use authenticated TLS SMTP connection

This PR will do no harm whenever it is merged, but I am waiting for a question to be answered by Cray I.T. before actually using authenticated SMTP in Jenkins.

**Comment by awallace-cray** [comment]
2018-10-23

Cray I.T. has made a change that allows us to use their SMTP server in this way. 

I would like to have @bradcray back in the office before activating this in Chapel, however.

**Comment by awallace-cray** [comment]
2018-10-24 merged PR #11360 after improvements suggested by mppf

**Comment by awallace-cray** [comment]
2018-10-25 
https://github.com/chapel-lang/chapel/pull/11360 was merged last night.

The related PRs are still available for review, as follows

> No change: https://github.com/chapel-lang/chapel/pull/11361
>            Remove all references to SMTP host "relaya"
> 
> No change: https://stash.us.cray.com/projects/CHAPEL/repos/chapel-code/pull-requests/415
>            Remove all references to SMTP host "relaya"
> 
> No change: https://stash.us.cray.com/projects/CHAPEL/repos/chapel-ci-config/pull-requests/585
>            Remove all references to SMTP host "relaya"
> 
> Rewritten: https://stash.us.cray.com/projects/CHAPEL/repos/chapel-ci-config/pull-requests/584
>       was: Inject chapel passwords (masked) into all jobs
>       now: Update the copy of send_email.py in dcsched to match chapel.git
> 

Updated:
    
The actual hostname and login will be defined as Global environment variables in the Jenkins server configuration (not the part covered by chapel-ci-config/JJB).

The password-file path is the send_email.py default value, except on the Cygwin slaves where it is an environment variable defined in the corresponding Slave configs.

**Comment by awallace-cray** [comment]
2018-10-26 

The e-mail restriction will be enforced starting Dec 1

**Comment by awallace-cray** [comment]
2018-10-27 23:30 : merged all PRs referenced above and restarted Jenkins

**Comment by awallace-cray** [comment]
2018-10-29 The cron mail conversion to authenticated SMTP did not go well.
See https://github.com/chapel-lang/chapel/pull/11493,
https://stash.us.cray.com/projects/CHAPEL/repos/chapel-code/pull-requests/417/

**Comment by awallace-cray** [comment]
2018-10-29 
Merged all 3 PRs above and re-ran correctness-test-c2chapel w the realtime mirror, not the nightly mirror, just to make sure cron mail would work. (Cron mail failed in the nightly build of this job, earlier this morning).

**Comment by awallace-cray** [comment]
Assuming all above changes work, one class of problem still remains which I did not recognize when I started this task.  Build notification emails sent from the Jenkins server itself (as opposed to cron mails which are sent from scripts running on the build slave). A few existing jobs have build notification emails which fire from the Jenkins configuration AND which attempt to send to a sourceforge list.  I have not been able to make either of the Jenkins email plug-ins that we use successfully connect to SMTP with authentication.  Failing that, we can just change the job configurations to 
(a) use the new SMTP server, unauthenticated, and 
(b) remove any sourceforge lists from the "To" or "Cc" fields in the JJB.
